Analyze coalesced mem

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #154768
* #153751
* #153748
* __->__ #153730
* #153723

Analyze memory expressions to see if they contain a coalescing symbol.


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

eellison committed:
Update

[ghstack-poisoned]

Skylion007 commented on torch/_inductor/tiling_utils.py:
```diff
@@ -29,6 +31,39 @@
+        variables[v] = 1
+        try:
+            new_val = sympy_subs(index, variables)
+        except ZeroDivisionError:
```
We don't want to log anything here?

eellison committed:
Update

[ghstack-poisoned]

etaf commented on test/inductor/test_loop_ordering.py:
```diff
@@ -741,6 +742,73 @@ def foo(x):
+
+            y_dtype = torch.float if not downcast_transposed_v else torch.float64
+            foo(
+                torch.rand(256, 256, device="cuda"),
```
Hi, May I suggest we mark this case as requires_cuda or replace the hardcode `cuda` with GPU_TYPE here? The `cuda` will  fail on XPU.

eellison committed:
Update on "Analyze coalesced mem"


Analyze memory expressions to see if they contain a coalescing symbol.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

jansel commented on torch/_inductor/tiling_utils.py:
```diff
@@ -196,7 +231,6 @@ def zip_equal(it1: Iterable[T], it2: Iterable[U]) -> Iterator[tuple[T, U]]:
         """
         if len(it1) != len(it2):
             raise ValueError(f"Lengths differ: {len(it1)} != {len(it2)}")
-        return zip(it1, it2)
```
Whoops?

eellison replied:
i think the inductor gpu tests are cuda 3.10.. will udpate, good catch

eellison committed:
Update on "Analyze coalesced mem"


Analyze memory expressions to see if they contain a coalescing symbol.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

eellison committed:
Update

[ghstack-poisoned]

eellison committed:
Update

[ghstack-poisoned]

eellison committed:
Update

[ghstack-poisoned]

eellison commented:
@pytorchbot merge

eellison committed:
Update

[ghstack-poisoned]
